src/vcl/FaissDescriptorSet.cc

```diff
@@ -339,3 +343,64 @@ long FaissIVFFlatDescriptorSet::add(float *descriptors, unsigned n,
 
   return id_first;
 }
+
+// FaissHNSWFlat
+// Note:
+// setting value of hnsw m= 48
+// M is number of connections each vertex will have
+// i.e. number of nearest neighbors that each vertex will connect to.
+// Total memory usage is (dim * 4 + M * 2 * 4) bytes per vector.
+
+// Rule of thumb to support 1M entries
+// according to
+// https://github.com/facebookresearch/faiss/wiki/Indexing-1M-vectors HNSW_M= 48
+// efConstruction = 2 x m
+// efSeach = 32
+// Default values from hnsw.h
+// int efConstruction = 40;
+// int efSearch = 16;
+// efsearch will be defined in the search params when IndexHNSW::search() is
+// called (can dynamically change during runtime) efconstruction is a compile
+// time paramerter will be defined in IndexHNSW.hnsw struct
+
+FaissHNSWFlatDescriptorSet::FaissHNSWFlatDescriptorSet(
+    const std::string &set_path)
```

Comment:
Can you add a C++ test where this function is used (set_path only)?

Reply:
fixed unit_tests/DescriptorSetReadFS_test.cc to call this function in the latest commit